src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -76,15 +77,185 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
+          if (liveState.reconciliationComplete()) {
+            reconcilePersistentState();
+            correctCurrentPhase();
+          }
+          break;
+        case JOURNAL_NODES :
+          if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+            correctCurrentPhase();
+          }
+          break;
+        case START_NAME_NODES :
+          if (liveState.getNameNodeSize() == (HDFSConstants.TOTAL_NAME_NODES)) {
+            // TODO move the reload to correctCurrentPhase and make it idempotent
+            reloadConfigsOnAllRunningTasks(driver);
+            correctCurrentPhase();
+          }
+          break;
+        case FORMAT_NAME_NODES :
+          if (!liveState.isNameNode1Initialized()
+              && !status.getMessage().equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)) {
+            sendMessageTo(
+                driver,
+                liveState.getFirstNameNodeTaskId(),
+                liveState.getFirstNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_INIT_MESSAGE);
+          } else if (liveState.isNameNode1Initialized()
+              && !liveState.isNameNode2Initialized()) {
+            sendMessageTo(
+                driver,
+                liveState.getSecondNameNodeTaskId(),
+                liveState.getSecondNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE);
+          } else if (liveState.isNameNode1Initialized()
+              && liveState.isNameNode2Initialized()) {
+            correctCurrentPhase();
+          }
+          break;
+        case DATA_NODES :
+          break;
+      }
+    } else {
+      log.warn(String.format("Don't know how to handle state=%s for taskId=%s",
+          status.getState(), status.getTaskId().getValue()));
+    }
+  }
+
+  @Override
+  public void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
+    log.info(String.format("Received %d offers", offers.size()));
+
+    if (liveState.getStagingTasksSize() != 0) {
+      log.info("Declining offers because tasks are currently staging");
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+    } else {
+      boolean acceptedOffer = false;
+      for (Offer offer : offers) {
+        if (acceptedOffer) {
+          driver.declineOffer(offer.getId());
+        } else {
+          switch (liveState.getCurrentAcquisitionPhase()) {
+            case RECONCILING_TASKS :
+              if (liveState.reconciliationComplete()) {
+                reconcilePersistentState();
+                log.info("Current persistent state:");
+                log.info(String.format("JournalNodes: %s", persistentState.getJournalNodes()));
+                log.info(String.format("NameNodes: %s", persistentState.getNameNodes()));
+                log.info(String.format("DataNodes: %s", persistentState.getDataNodes()));
+                correctCurrentPhase();
+                driver.declineOffer(offer.getId());
```

Comment:
It first needs to transition in the state machine. It will correct the phase and move from RECONCILE_TASKS to LAUNCH_JOURNAL_NODES and then in the next iteration it will accept the offer.